### PR TITLE
Unit Tests for hub.go & channel.go

### PR DESCRIPTION
**Pull request by mermel**
Basic unit testing for methods around hub and channel.

Please let me know if there is a standardized format for test failure messages, for the most part I just printed the expectation and output.  

I chose to be super explicit when setting up the cases ie: "h.subscribe(command{cmd: SUBSCRIBE, path: "/monkey", conn: newTestConnection()})" to make it clear which events are triggering each action, but maybe it's overkill...

Depending on how the other tests go, I may modify this implementation.

### CODE REVIEW

File: hub_test.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	h.subscribe(command{cmd: SUBSCRIBE, path: "/banana", conn: newTestConnection()})
+	if len(h.channels) != 2 {
+		t.Fatal("Expectation: 2, Received:", len(h.channels))
+	}
+}
+
+func TestPublish(t *testing.T) {
+	h := newHub()
+	// Publishing to a non-existant channel
+	// should drop meessage
```
Comment (gravityrail) [inline hub_test.go]: tyypo ;)

File: channel_test.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	c.publish([]byte("banana"))
+
+	text1, text2 := <-conn.send, <-conn2.send
+	if string(text1) != "banana" || string(text2) != "banana" {
+		t.Fatal("Expectation: published text for connections should be 'banana', Received:", string(text1), string(text1))
+	}
+
+	//Empty string should not publish
+	c.publish([]byte(""))
+	if len(conn.send) != 0 {
```
Comment (skeltoac) [inline channel_test.go:50]: Is there a race here?

File: channel_test.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	conn := newTestConnection()
+	c.subscribe(conn)
+
+	c.unsubscribe(conn)
+	if len(c.connections) != 0 {
+		t.Fatal("Expectation: 0, Received:", len(c.connections))
+	}
+
+	defer func() {
+		if r := recover(); r == nil {
```
Comment (skeltoac) [inline channel_test.go:67]: Clever!

File: channel_test.go (hunk 2 above)
Comment (skeltoac) [inline channel_test.go:50]: Oh I see. The conn is not running. :+1:

**Pull request merged**